test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -2307,6 +2335,11 @@ private static final class NodeNamePredicate implements Predicate<NodeAndClient>
         public boolean test(NodeAndClient nodeAndClient) {
             return nodeName.equals(nodeAndClient.getName());
         }
+
+        @Override
+        public String toString() {
+            return "name: " + nodeName;
+        }
     }
```

Comment:
NIT: I think this one could be a record providing free toString implementation

Reply:
++ good call, thanks